nupic/research/frameworks/pytorch/models/common_models.py

```diff
@@ -119,12 +119,30 @@ def conv_block(cls, in_channels, out_channels, kernel_size, stride, padding):
         ]
 
     @property
-    def fast_params(self):
-        return self.adaptation.parameters()
+    def named_fast_params(self):
+        """
+        Returns named params of adaption network, being sure to prepend
+        the names with "adaptation."
+        """
+        named_parameters = self.adaptation.named_parameters()
```

Comment:
Why don't we replace `adaptation` and `representation` with `fast` and `slow`, respectively, everywhere? That may make things simpler.

Reply:
The `fast` and `slow` scheme was helpful for us in defining the `MetaContinualLearningExperiment`, but I'm not sure it makes sense to apply it broadly. Here, the `OMLNetwork` is implemented to mimic that of the papers. As one reads the code, they can make the connection mentally to the paper.

In contrast, `named_fast_params` and `named_slow_params` serves as an interface and helps the model interact with the experiment class.

Note, we'll always need such an interface. It helps keep the experiment agnostic to the model. Imagine if we have a network with multiple dendritic segments. We may not be able to divide our modules cleanly into `self.slow` and `self.fast`. Thus, we can allow the model to be defined in whatever way it needs to be, as long as it includes the interface.

Reply:
I see your point though `named_fast_params` is a bit cumbersome. I can think of a couple of other ways to do it, but I'm not sure they're any better.

Reply:
I agree with Karan, keep the nomenclature stable across the framework. The link to adaptation / representation can be made clear in the docstring instead for potential readers/users.

But you certainly gave a lot more thought to this, so however you think it's best.

Reply:
I agree to keep the nomenclature consistent. However, I think it's important to have two separate classifications: one for our modules and one for the params. The `OMLNetwork` has `adaptation` which has all of the fast params and `representation` which has all the slow params, but maybe there won't always be a perfect overlap between the two. That's my concern. We can keep the naming in mind though as we move forward, and change it if we see it's too confusing.